tests/unit/mappers/splitMapper.test.ts

```diff
@@ -0,0 +1,96 @@
+            {inValue: ';asd.dsa;',  inSeparator: ';', expectedResult: ['','asd.dsa','']},
+            {inValue: ',',  inSeparator: ';', expectedResult: [',']},
+            {inValue: '0,0',  inSeparator: ';', expectedResult: ['0,0']},
+            {inValue: 'asd,dsa',  inSeparator: ';', expectedResult: ['asd,dsa']},
+            {inValue: ',asd,dsa',  inSeparator: ';', expectedResult: [',asd,dsa']},
+            {inValue: ',asd,dsa,',  inSeparator: ';', expectedResult: [',asd,dsa,']},
+            {inValue: ',asd;dsa,',  inSeparator: ';', expectedResult: [',asd','dsa,']},
+            {inValue: '[1,2,3]',  inSeparator: ';', expectedResult: ['[1,2,3]']},
+            {inValue: 'Ala ma kota, a kot ma Ale',  inSeparator: ', a ', expectedResult: ['Ala ma kota', 'kot ma Ale']},
+            {inValue: 'bbbBBBbbbBBBaaa',  inSeparator: 'BBB', expectedResult: ['bbb', 'bbb', 'aaa']},
+
+            // invalid inputs won't to check
+            // {inValue: {}, expectedResult: ... },
+        ];
+        dataProvider.forEach(({inValue,inSeparator, expectedResult}) => {
+            it(`splitMapper with separator ${inSeparator} for input "${inValue}" SHOULD return "${JSON.stringify(expectedResult)}"`, () => {
+                const mapper = splitMapper.separator(inSeparator);
+                chai.expect(mapper(inValue)).eql(expectedResult);
+            });
+        });
+    });
+    describe('splitMapper custom mapper', () => {
+        const dataProvider = [
+            {inValue: 'Ala,ma,kota,,Kot,ma,Ale', inMapper:upperCaseMapper, expectedResult: ['ALA','MA','KOTA','','KOT','MA','ALE']},
+            {inValue: 'Ala,ma,kota,,Kot,ma,Ale', inMapper:lowerCaseMapper, expectedResult: ['ala','ma','kota','','kot','ma','ale']},
+            {inValue: 'Ala,ma,kota,,Kot,ma,Ale', inMapper:isEmpty, expectedResult: [false,false,false,true,false,false,false]},
+            {inValue: 'Ala,ma,kota,,Kot,ma,Ale', inMapper:(v:string) => v.length, expectedResult: [3,2,4,0,3,2,3]},
+        ];
+        dataProvider.forEach(({inValue,inMapper, expectedResult}) => {
+            it(`splitMapper with separator ${inMapper.constructor.name} for input "${inValue}" SHOULD return "${JSON.stringify(expectedResult)}"`, () => {
+                const mapper = splitMapper.itemMapper<unknown>(inMapper);
+                chai.expect(mapper(inValue)).eql(expectedResult);
+            });
+        });
+    });
+    describe('splitMapper complex / advanced', () => {
+        it(`Map sentence into matrix word in sentence`, () => {
+            // assumptions
+            const input = 'Lorem ipsum dolor sit amet. consectetur adipiscing elit. Nullam placerat massa nec efficir. ';
+            const expectedResult = [
+                ['Lorem', 'ipsum', 'dolor', 'sit', 'amet'],
+                ['consectetur', 'adipiscing', 'elit'],
+                ['Nullam', 'placerat', 'massa', 'nec', 'efficir'],
+                ['']
+            ]
+
+            // building a mapper
```

Comment:
Just an idea: Maybe you could add a function `buildMapper()` which returns the wordsInSentencesMapper, so you would not need the comment.

Reply:
I always prefer self-descriptive function than comments, however it is test case code and I wouldn’t make it more complicated. Wdyt?

Reply:
Well, it's just a matter of taste and it always depends. I tend to always extract a method, if there is a comment.

Reply:
done 855d198